pinot-core/src/main/java/org/apache/pinot/core/startree/StarTreeUtils.java

```diff
@@ -179,4 +183,51 @@ public static boolean isFitForStarTree(StarTreeV2Metadata starTreeV2Metadata,
     // Check predicate columns
     return starTreeDimensions.containsAll(predicateColumns);
   }
+
+  /**
+   * Evaluates a given filter to ascertain if the OR clause is valid for StarTree processing.
+   *
+   * StarTree supports OR predicates on a single dimension only (d1 < 10 OR d1 > 50).
+   */
+  private static boolean isOrClauseValidForStarTree(FilterContext filterContext) {
+    assert filterContext != null;
+
+    Set<String> seenLiterals = new HashSet<>();
+
+    isOrClauseValidForStarTreeInternal(filterContext, seenLiterals);
+
+    return seenLiterals.size() == 1;
+  }
+
+  /** Internal processor for the above evaluator */
+  private static void isOrClauseValidForStarTreeInternal(FilterContext filterContext, Set<String> seenLiterals) {
+    assert filterContext != null;
+
+    if (filterContext.getType() == FilterContext.Type.OR) {
+      List<FilterContext> childFilterContexts = filterContext.getChildren();
+
+      for (FilterContext childFilterContext : childFilterContexts) {
+        isOrClauseValidForStarTreeInternal(childFilterContext, seenLiterals);
+      }
+    } else if (filterContext.getType() == FilterContext.Type.PREDICATE) {
+      String literalValue = validateExpressionAndExtractLiteral(filterContext.getPredicate());
+
+      if (literalValue != null) {
```

Comment:
When the lhs is not an identifier, we should fail the check because the OR clause is not valid